inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c

```diff
@@ -3146,41 +3139,36 @@ ncStatus_t ncFifoDestroy(struct ncFifoHandle_t ** fifoHandle)
         if (XLinkWriteData(handle->streamId, (uint8_t *) & msg, sizeof(msg)) !=
             0) {
             mvLog(MVLOG_ERROR, "Failed to write to fifo before deleting it!");
-            return NC_ERROR;
+            rc = NC_ERROR;
         }
     }
 
-    ncStatus_t rc = NC_OK;
     graphCommonCommand_t cmd;
     cmd.type = GRAPH_BUFFER_DEALLOCATE_CMD;
     cmd.id = handle->id;
 
     struct _devicePrivate_t *d = handle->dev;
     CHECK_MUTEX_SUCCESS(pthread_mutex_lock(&d->graph_stream_m));
     rc = trySendCommand(d->graph_monitor_stream_id, &cmd, sizeof(cmd));
-    if(rc != 0){
+    if (rc != 0) {
         CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->graph_stream_m));
         mvLog(MVLOG_WARN, "can't send command\n");
-        return rc;
-    }
-    if (checkGraphMonitorResponse(d->graph_monitor_stream_id)) {
+        rc = NC_ERROR;
+    } else if (checkGraphMonitorResponse(d->graph_monitor_stream_id)) {
         CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->graph_stream_m));
         mvLog(MVLOG_WARN, "myriad NACK\n");
-        return NC_ERROR;
+        rc = NC_ERROR;
     }
     CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->graph_stream_m));
 
     CHECK_MUTEX_SUCCESS(pthread_mutex_lock(&d->dev_data_m));
     if (deallocateFifo(handle)) {
         CHECK_MUTEX_SUCCESS(pthread_mutex_unlock(&d->dev_data_m));
```

Comment:
Why not deleted like [here](https://github.com/openvinotoolkit/openvino/pull/8509/files#:~:text=mvLog(MVLOG_WARN%2C%20%22failed%20deallocateFifo%5Cn%22)%3B) ?

Reply:
And in some other places.

Reply:
I agree with you. Changed